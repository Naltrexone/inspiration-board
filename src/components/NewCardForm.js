import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {

  constructor(props) {
    super(props);

    const baseState = {
      text: '',
      emoji: '',
      errorMessages: []
    };

    this.state = {...baseState};
    this.baseState = {...baseState};
  }

  onInputChange = (e) => {
    const { name, value } = e.target;

    const newState = {};
    newState[name] = value;
    this.setState(newState);
  }

  onFormSubmit = (e) => {
    e.preventDefault();

    if (this.state.text === '') {
      this.setState({ errorMessages: [...this.state.errorMessages, "text cannot be blank"]});
      return;
    }

    const newCard = {...this.state};
    this.setState(this.baseState);
    this.props.addCard(newCard);

    this.setState({
      errorMessages: []
    });
  }


  render () {

    const emojiList = EMOJI_LIST.map((emojiText, i) => {

      const emojiCode = emoji.getUnicode(emojiText);

      return (<option value = {emojiCode} key = {i}>{emojiCode}</option>)
    });

    const errorMessages = this.state.errorMessages.map((message, i) => {
      return <li key={i}>{message}</li>;
      });

    return(
      <div className="new-card-form">
        <section className="new-card-form__header">
          New Card Form
          <ul>
            {errorMessages}
          </ul>
        </section>

        <form
          onSubmit={this.onFormSubmit}
          >
          <div className="new-card-form__form">
            <label htmlFor="text" className="new-card-form__form-label">Text:</label>
            <textarea
              name="text"
              value={this.state.text}
              onChange={this.onInputChange}
              className="new-card-form__form-textarea"
              />
          </div>
          <div className="new-card-form__form">
            <label htmlFor="emoji" className="new-card-form__form-label">Emoji:</label>
            <select name="emoji" value={this.state.emoji} onChange={this.onInputChange}
              className="new-card-form__form-select">
              {emojiList}
            </select>
          </div>
          <input
            type="submit"
            value="Add Card"
            className="new-card-form__form-button"
            />
        </form>
      </div>
    )
  }
}

NewCardForm.propTypes = {
  addCard: PropTypes.func,
};

export default NewCardForm;
