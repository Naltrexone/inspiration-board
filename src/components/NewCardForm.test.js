import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NewCardForm from './NewCardForm';

describe('NewCardForm', () => {
  test('That it matches an existing snapshot', () => {
    const cardForm = shallow(
      <NewCardForm addCardCallback={() => {} } />
    );

    expect(cardForm).toMatchSnapshot();

    cardForm.unmount();
  });
});
