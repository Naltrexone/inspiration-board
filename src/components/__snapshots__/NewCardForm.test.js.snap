exports[`NewCardForm That it matches an existing snapshot 1`] = `
<form
  className="new-card-form new-card-form__form"
  onSubmit={[Function]}
>
  <h3
    className="new-card-form__header"
  >
    Add a new inspirational note to this board.
  </h3>
  <div>
    <label
      className="new-card-form__form-label"
      htmlFor="text"
    >
      Message
    </label>
    <textarea
      className="new-card-form__form-textarea"
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      className="new-card-form__form-label"
      htmlFor="emoji"
    >
      Emoji (optional)
    </label>
    <select
      className="new-card-form__form-select"
      name="emoji"
      onChange={[Function]}
      value=""
    >
      <option
        key="0"
        value=""
      />
      <option
        key="1"
        value="heart_eyes"
      >
        😍
      </option>
      <option
        key="2"
        value="beer"
      >
        🍺
      </option>
      <option
        key="3"
        value="clap"
      >
        👏
      </option>
      <option
        key="4"
        value="sparkling_heart"
      >
        💖
      </option>
      <option
        key="5"
        value="heart_eyes_cat"
      >
        😻
      </option>
      <option
        key="6"
        value="dog"
      >
        🐶
      </option>
    </select>
  </div>
  <div>
    <input
      className="new-card-form__form-button"
      type="submit"
    />
  </div>
</form>
`;
